compilation error on master: irrefutable pattern (Windows only)

**Describe the bug**

commit `38e3dcb98c60a239064653cc002f5c458b3eab67` does not compile with rustc `1.31.0` (`abe02cefd` 2018-12-04).

```
error[E0162]: irrefutable if-let pattern
  --> src\network\grpc\bootstrap.rs:95:16
   |
95 |         if let Connection::Tcp(addr) = self.connection {
   |                ^^^^^^^^^^^^^^^^^^^^^ irrefutable pattern
```

**To Reproduce**

1. checkout the commit;
2. make sure cardano-deps is at the right commit too;
3. `cargo build`

**Expected behavior**

to compile without errors

**Additional context**

rust toolchain used here: `stable-x86_64-pc-windows-msvc`

--- a/src/network/grpc/bootstrap.rs
+++ b/src/network/grpc/bootstrap.rs
@@ -92,10 +92,14 @@ impl tokio_connect::Connect for Peer {
     type Future = ConnectFuture;
 
     fn connect(&self) -> Self::Future {
-        if let Connection::Tcp(addr) = self.connection {
-            TcpStream::connect(&addr)
-        } else {
-            unimplemented!()
+        match &self.connection {
+            Connection::Tcp(ref addr) => {
+                TcpStream::connect(addr)
+            }
+            #[cfg(unix)]
+            Connection::Unix(_) => {
+                unimplemented!()
+            }
         }
     }
 }

